environs/manual/sshprovisioner/sshprovisioner.go

```diff
@@ -115,21 +115,26 @@ func detectSeriesAndHardwareCharacteristics(host string) (hc instance.HardwareCh
 		if stderr.Len() != 0 {
 			err = fmt.Errorf("%v (%v)", err, strings.TrimSpace(stderr.String()))
 		}
-		return hc, "", err
+		return hc, base, err
 	}
 	lines := strings.Split(stdout.String(), "\n")
-	series = strings.TrimSpace(lines[0])
+	os := strings.TrimSpace(lines[0])
+	channel := strings.TrimSpace(lines[1])
+	base, err = corebase.ParseBase(os, channel)
+	if err != nil {
+		return hc, base, err
+	}
 
-	arch := arch.NormaliseArch(lines[1])
+	arch := arch.NormaliseArch(lines[2])
```

Comment:
How can the arch position change in `lines` since the script output is the same?